agent/api/ecsclient/client.go

```diff
@@ -11,98 +11,48 @@
 // express or implied. See the License for the specific language governing
 // permissions and limitations under the License.
 
-package api
+package ecsclient
 
 import (
 	"errors"
 	"net/http"
 	"runtime"
 	"time"
 
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/async"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/ec2"
 	"github.com/aws/amazon-ecs-agent/agent/ecs_client/model/ecs"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/cihub/seelog"
 	"github.com/docker/docker/pkg/system"
+)
 
-	"github.com/aws/amazon-ecs-agent/agent/config"
-	"github.com/aws/amazon-ecs-agent/agent/ec2"
-	"github.com/aws/amazon-ecs-agent/agent/logger"
-	"github.com/aws/amazon-ecs-agent/agent/utils"
+const (
+	ecsMaxReasonLength    = 255
+	pollEndpointCacheSize = 1
+	pollEndpointCacheTTL  = 10 * time.Minute
```

Comment:
My thoughts here are that anything >= `10m` should be good enough here. I can increase it to `30m`. What value do you have in your mind?